src/modules/shlvl.rs

```diff
@@ -0,0 +1,54 @@
+use std::env;
+
+use super::{Context, Module, SegmentConfig};
+
+use crate::config::RootModuleConfig;
+use crate::configs::shlvl::ShLvlConfig;
+
+const SHLVL_ENV_VAR: &str = "SHLVL";
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let shlvl = get_shlvl_value();
```

Comment:
You can remove the match below if you use the `?` operator, i.e.:
`let shlvl = get_shlvl_value()?;`

Reply:
thanks. for my own edification, does that essentially destructure the `Option` and return `None` if the inner one is?

Reply:
Yes.

Think of `x?` like:
`match x { Some value => value, None => return None }`

It's a little subtle, because its effect encompasses the local function, not just the current expression.

This is why you can only use the syntax with a function which returns `Option<T>`, or a function which returns `Result<T, E>` (The de-sugaring is slightly different for result functions, as you might imagine).